beautifying index page

dimitrovmaksim opened the pull request:
Beautifying the index page

antonrd commented on app/assets/javascripts/application.js:
```diff
@@ -10,8 +10,10 @@
+// = require bootstrap-sprockets
+// = require_tree .
+// = require bootstrap
+// = require bootstrap.min
```
Не ти трябват и двата файла, единия би трябвало да е минифицирана версия на другия. Сложи само bootstrap файла, рейлс ще се погрижи да го минифицира в production.

antonrd commented on app/assets/stylesheets/application.scss:
```diff
@@ -2,3 +2,5 @@
 @import "bootstrap";
 @import "*";
 
+
```
Ненужни нови редове

antonrd commented on app/assets/stylesheets/articles.scss:
```diff
@@ -1,3 +1,5 @@
 // Place all the styles related to the Articles controller here.
 // They will automatically be included in application.css.
 // You can use Sass (SCSS) here: http://sass-lang.com/
+
```
Ненужни нови редове

antonrd commented on app/assets/stylesheets/css/blog-home.css:
```diff
@@ -0,0 +1,13 @@
+
+footer {
+    margin: 50px 0;
+}
```
Добавяй нов ред в края на всеки файл. Смята се за добра практика. Би трябвало да имаш настройка в едитора.

antonrd commented on app/views/articles/index.html.erb:
```diff
@@ -1,51 +1,88 @@
+       <div class="row">
+          <div class="col-lg-6">
+            <ul class="list-unstyled">
+              <li><a href="#">October</a></li>
```
Тези ми се вижда, че не бива да са така хард-код-нати. Ами да се извличат на база какви статии има в базата. Или това е само временно?

dimitrovmaksim replied:
Временно е за изглед

antonrd commented:
Този ПР добавя и user към article, което не е описано в ПР-а, а би трябвало, за да е ясно какво има в този ПР.

dimitrovmaksim commented:
UI tweaks to the Index page:

1. Added "user_id" column to the "articles" database. Now author's name will be shown on the index page on article creation.
2. Added "image_url" column to the "articles" database. Now pictures can be added to the articles. Accepted file extensions: jpg, gif and png. Images should be placed in "app/assets/images". Usage: 
![image_use](https://cloud.githubusercontent.com/assets/20456492/20382723/c2f7cf44-acb5-11e6-9ac8-b56b291c84dd.PNG)

3. Various Index UI tweaks using templates from : https://startbootstrap.com/template-categories/blogs/

Screenshot:
![index_screenshot](https://cloud.githubusercontent.com/assets/20456492/20382415/8d13fe6c-acb4-11e6-8832-f6f72024b03b.PNG)

antonrd commented on LICENSE:
```diff
@@ -0,0 +1,22 @@
+
+The MIT License (MIT)
+
+Copyright (c) 2013-2016 Blackrock Digital LLC.
```
Това име на компания ли е?

antonrd commented on app/views/articles/index.html.erb:
```diff
@@ -23,7 +21,7 @@
           </p>
 
-            <img class="img-responsive" src="http://placehold.it/500x150" alt="">
+            <%= image_tag(article.image_url, :class => "img-responsive", size: "350x150") %>
```
`image_url` май може да е `nil` понякога? в тези случаи мисля, че не трябва да се генерира тук картинка

antonrd commented:
Файла `app/assets/stylesheets/css/bootstrap.min.css` не е нужен в проекта.

antonrd commented:
Като цяло за картинките не е практично да се качват само в кода на самия проект. Причината е че тогава за нов пост с картинка, човек ще трябва да я слага в кода, да пуш-ва някъде, да диплойва новата версия и т.н. Би могло да се позволява и външни линкове да се слагат, а в бъдеще и да се качват картинки. Но нека засега е така, за да не усложняваме нещата.

dimitrovmaksim commented:
1. Fix: Now if "image_url" is nil, no image is generated.
2. Deleted files that are not needed.
